trie/hexary.py

```diff
@@ -781,15 +787,18 @@ def __contains__(self, key):
     def squash_changes(self):
         scratch_db = ScratchDB(self.db)
         with scratch_db.batch_commit(do_deletes=self.is_pruning):
-            memory_trie = type(self)(scratch_db, self.root_hash, prune=True)
+            Trie = type(self)
+            memory_trie = Trie(scratch_db, self.root_hash, prune=True, ref_count=self._ref_count)
             yield memory_trie
 
         if self.root_hash != memory_trie.root_hash:
             try:
-                self._set_root_node(memory_trie.root_node.raw)
```

Comment:
This method had too much pruning-related baggage. For this situation, we need to manage things more directly, so just use `_set_raw_node` below.